src/relay/pass/simplify_inference.cc

```diff
@@ -0,0 +1,77 @@
+/*!
+ * Copyright (c) 2018 by Contributors
+ * \file simplify_inference.cc
+ */
+#include <tvm/relay/pass.h>
+#include <tvm/relay/expr_functor.h>
+#include <tvm/relay/attrs/nn.h>
+#include "./pattern_util.h"
+
+namespace tvm {
+namespace relay {
+
+Expr BatchNormToInferUnpack(const Attrs attrs,
+                            Expr data,
+                            Expr gamma,
+                            Expr beta,
+                            Expr moving_mean,
+                            Expr moving_var) {
+  const auto param = attrs.as<BatchNormAttrs>();
+  Expr epsilon = MakeConstantScalar(Float(32), static_cast<float>(param->epsilon));
+  Expr var_add_eps = Add(moving_var, epsilon);
+  Expr sqrt_var = Sqrt(var_add_eps);
+  Expr scale = Divide(MakeConstantScalar(Float(32), 1.0f), sqrt_var);
+
+  if (param->scale) {
+    scale = Multiply(scale, gamma);
+  }
+  Expr neg_mean = Negative(moving_mean);
+  Expr shift = Multiply(neg_mean, scale);
+  if (param->center) {
+    shift = Add(shift, beta);
+  }
+
+  int axis = param->axis;
+  const auto* tdata = data->type_as<TensorTypeNode>();
+  scale = ExpandBiasToMatchAxis(scale, tdata->shape.size(), {axis});
+  shift = ExpandBiasToMatchAxis(shift, tdata->shape.size(), {axis});
+
+  Expr out = Multiply(data, scale);
+  out = Add(out, shift);
+  return out;
+}
+
+class Simplifier : public ExprMutator {
```

Comment:
Simplifier->InferenceSimplifier, because this likely will affect other class names(imagine if we have another simplifier in the same namespace)